src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -22,7 +22,7 @@ internal partial class DocumentState : TextDocumentState
 
         private readonly HostLanguageServices _languageServices;
         private readonly ParseOptions _options;
-
+        private readonly ValueSource<AnalyzerConfigSet> _analyzerConfigSetSource;
```

Comment:
fwiw, i don't like the name AnalyzerConfigSet.

Reply:
Any better suggestions? We're not fans either but didn't have many other ideas.

Reply:
Is this set the unique set of all analyzer configurations available or the configurations that apply to this document? 

If the latter I would just call it `AnalyzerConfigurations` or `AnalyzerConfigs`

Reply:
It's the set of all configurations available, with methods that let you figure out how to combine them.